Framework/API/src/NearestNeighbours.cpp

```diff
@@ -141,18 +126,16 @@ void NearestNeighbours::build(const int noNeighbours) {
   BoundingBox bbox;
   // Base the scaling on the first detector, should be adequate but we can look
   // at this
-  IDetector_const_sptr firstDet = (*spectraDets.begin()).second;
-  firstDet->getBoundingBox(bbox);
+  const auto &firstDet = m_spectrumInfo.detector(indices.front());
```

Comment:
I think it actually beyond the scope of this PR, however I've seen a lot of code that uses the first detector to get information on the group. I think that's another discussion to have at a later point.